src/Query/DefaultQueryBuilder.php

```diff
@@ -0,0 +1,194 @@
+            $defaultValues,
+        );
+    }
+
+    public function writeDataQueryStatement(
+        Connection $connection,
+        string $tableName,
+        ExportConfig $exportConfig,
+    ): string {
+        $query = <<<SQL
+LOAD DATA LOCAL INFILE '%s'
+INTO TABLE %s
+CHARACTER SET utf8
+FIELDS TERMINATED BY ','
+OPTIONALLY ENCLOSED BY '\"'
+ESCAPED BY ''
+IGNORE 1 LINES
+SQL;
+
+        return sprintf(
+            $query,
+            $exportConfig->getTableFilePath(),
+            $connection->quoteIdentifier($tableName),
+        );
+    }
+
+    public function tableExistsQueryStatement(Connection $connection, string $tableName): string
+    {
+        return sprintf(
+            'SELECT * FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_NAME = %s',
+            $connection->quote($tableName),
+        );
+    }
+
+    public function upsertUpdateRowsQueryStatement(
+        Connection $connection,
+        ExportConfig $exportConfig,
+        string $stageTableName,
+    ): string {
+        $columns = array_map(function ($item) {
+            return $item->getDbName();
+        }, $exportConfig->getItems());
+
+        // update data
+        $joinClauseArr = array_map(fn($item) => sprintf(
+            'a.%s = b.%s',
+            $connection->quoteIdentifier($item),
+            $connection->quoteIdentifier($item),
+        ), $exportConfig->getPrimaryKey());
+        $joinClause = implode(' AND ', $joinClauseArr);
+
+        $valuesClauseArr = array_map(fn($item) => sprintf(
+            'a.%s = b.%s',
+            $connection->quoteIdentifier($item),
+            $connection->quoteIdentifier($item),
+        ), $columns);
+        $valuesClause = implode(',', $valuesClauseArr);
+
+        return sprintf(
+            'UPDATE %s a INNER JOIN %s b ON %s SET %s;',
+            $connection->quoteIdentifier($exportConfig->getDbName()),
+            $connection->quoteIdentifier($stageTableName),
+            $joinClause,
+            $valuesClause,
+        );
+    }
+
+    public function upsertDeleteRowsQueryStatement(
+        Connection $connection,
+        ExportConfig $exportConfig,
+        string $stageTableName,
+    ): string {
+        $joinClauseArr = array_map(fn($item) => sprintf(
+            'a.%s = b.%s',
+            $connection->quoteIdentifier($item),
+            $connection->quoteIdentifier($item),
+        ), $exportConfig->getPrimaryKey());
+        $joinClause = implode(' AND ', $joinClauseArr);
+
+        return sprintf(
+            'DELETE a.* FROM %s a INNER JOIN %s b ON %s',
+            $connection->quoteIdentifier($stageTableName),
+            $connection->quoteIdentifier($exportConfig->getDbName()),
+            $joinClause,
+        );
+    }
+
+    public function upsertQueryStatement(
+        Connection $connection,
+        ExportConfig $exportConfig,
+        string $stageTableName,
+    ): string {
+        $columns = array_map(function ($item) use ($connection) {
```

Comment:
`$quotedDbColumnNames`? Zas ať je to clear na první pohled?

Reply:
ok :)